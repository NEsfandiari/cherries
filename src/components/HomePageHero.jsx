import React from 'react'
import styled from 'styled-components'
import { StaticQuery, graphql } from 'gatsby'
import { HomePageButton } from './atoms'

const Container = styled.div`
  position: relative;
  #hero-image {
    width: 100vw;
    height: 431.97px;
    object-fit: cover;
  }
  #hero-image-text-container {
    position: absolute;
    top: 30%;
    left: 13%;
    width: 308px;
  }
  #hero-image-text {
    font-size: 26px;
    color: #e20031;
    width: 100%;
    margin: 15px 0px;
    padding: 0px 10px;
  }

  @media (max-width: 420px) {
    #hero-image-text-container {
      position: static;
      margin: 0 auto;
      display: flex;
      flex-direction: column;
      align-items: center;
    }
    #hero-image-text {
      text-align: center;
    }
  }
`
// This is for testing purposes only!
export const PureHomePageHero = ({ data }) => (
  <Container>
    <img
      id="hero-image"
      src={data.contentfulHomePage.contentfulData.heroImage.file.url}
      alt="hero"
    />
    <div id="hero-image-text-container">
      <h4 id="hero-image-text">
        {data.contentfulHomePage.contentfulData.heroImageText}
      </h4>
      <HomePageButton />
    </div>
  </Container>
)

export default () => (
  <StaticQuery
    query={graphql`
      {
        contentfulHomePage(pageTitle: { eq: "Home Page" }) {
          heroImage {
            file {
              url
            }
          }
          heroImageText
        }
      }
    `}
    render={data => {
      const contentfulData = data.contentfulHomePage
      const heroImage = contentfulData.heroImage.file.url
      const heroImageText = contentfulData.heroImageText

      return (
        <Container>
          <img id="hero-image" src={heroImage} alt="hero" />
          <div id="hero-image-text-container">
            <h4 id="hero-image-text">{heroImageText}</h4>
            <HomePageButton />
          </div>
        </Container>
      )
    }}
  />
)
