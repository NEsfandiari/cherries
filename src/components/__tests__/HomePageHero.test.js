import React from 'react'
import renderer from 'react-test-renderer'
import { HomePageHero, PureHomePageHero } from '../HomePageHero'

describe('HomePageHero', () => {
  it('renders correctly', () => {
    const data = {
      contentfulHomePage: {
        contentfulData: {
          heroImage: {
            file: {
              url:
                '//images.ctfassets.net/4yejk6rho2rp/2Idi2Gr1KMYwYUQ8qWSqYq/b4406e872c773f97f8c0a188b7b608d9/homepage-hero-image.png',
            },
          },
          heroImageText: 'Fresh earrings, hand-picked by you',
        },
      },
    }
    const tree = renderer.create(<PureHomePageHero data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
