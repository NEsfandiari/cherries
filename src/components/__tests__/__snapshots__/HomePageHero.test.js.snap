// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageHero renders correctly 1`] = `
<div
  className="sc-jTzLTM chmQRN"
>
  <img
    alt="hero"
    id="hero-image"
    src="//images.ctfassets.net/4yejk6rho2rp/2Idi2Gr1KMYwYUQ8qWSqYq/b4406e872c773f97f8c0a188b7b608d9/homepage-hero-image.png"
  />
  <div
    id="hero-image-text-container"
  >
    <h4
      id="hero-image-text"
    >
      Fresh earrings, hand-picked by you
    </h4>
    <div
      className="sc-bwzfXH joayuL"
    >
      <a
        href="/subscribe"
      >
        GET STARTED
      </a>
    </div>
  </div>
</div>
`;
